sender_test.go

```diff
@@ -0,0 +1,25 @@
+package enmime_test
+
+import (
+	"net/smtp"
+	"strings"
+	"testing"
+
+	"github.com/jhillyerd/enmime"
+)
+
+func TestSMTPSend(t *testing.T) {
+	from := "user@example.com"
+	auth := smtp.PlainAuth("", from, "password", "mail.example.com")
+	s := enmime.NewSMTP("0.0.0.0", auth)
+
+	// Satisfy requirements of the Send method and use an intentionally malformed From Address to
+	// elicit an expected error from smtp.SendMail, which can be type-checked and verified.
+	err := s.Send(from+"\rinvalid", []string{"to@example.com"}, []byte("message content"))
```

Comment:
Should also validate that `\n` is treated as an invalid character.

Reply:
This code isn't really interested in what is invalid, it's just testing that Go's built-in Send is actually called.  It's just a refactored version of what @requaos did in #138 -- bit of a hack to maximize test coverage without being able to mock.